tests/js/subscription-tests.js

```diff
@@ -328,4 +380,116 @@ module.exports = {
+                const sub = query.subscribe("a_objs" );
+                sub.addListener((subscription, state) => {
+                    if (state === Realm.Sync.SubscriptionState.Complete) {
+                        sub.removeAllListeners();
+                        let a_objects = realm.objects('ObjectA');
+                        let parent_objects = realm.objects('Parent');
+                        TestCase.assertEqual(a_objects.length, 2);
+                        // parent objects were removed from local copy because they were not subscribed to
+                        TestCase.assertEqual(parent_objects.length, 0);
+                        resolve();
+                    }
+                    else if (state === Realm.Sync.SubscriptionState.Error) {
+                        reject(subscription.error);
+                    }
+                });
+            });
+        });
+    },
+
+    testSubscribeParentsWithForwardLinks() {
+        if (!isNodeProccess) {
+            return;
+        }
+
+        return getRealm().then(realm => {
+            realm.write(() => {
+                let obj_a1 = realm.create('ObjectA', {name: "a1"});
+                let obj_a2 = realm.create('ObjectA', {name: "a2"});
+                let parent_1 = realm.create('Parent', {name: "p1", child: obj_a1});
+                let parent_2 = realm.create('Parent', {name: "p2", child: obj_a1});
+            });
+            return new Promise((resolve, reject) => {
+                const query = realm.objects("Parent");
+                const sub = query.subscribe("parent_objects" );
+                sub.addListener((subscription, state) => {
+                    if (state === Realm.Sync.SubscriptionState.Complete) {
+                        sub.removeAllListeners();
+                        let parent_objects = realm.objects('Parent');
+                        let a_objects = realm.objects("ObjectA");
+                        TestCase.assertEqual(parent_objects.length, 2);
+                        // one object is not linked to and therefore not included in the subscription
+                        TestCase.assertEqual(a_objects.length, 1);
+                        resolve();
+                    }
+                    else if (state === Realm.Sync.SubscriptionState.Error) {
+                        reject(subscription.error);
+                    }
+                });
+            });
+        });
+    },
+
+    testSubscribeToChildrenWithNamedParents() {
+        if (!isNodeProccess) {
+            return;
+        }
+        return verifySubscriptionWithParents("parents");
+    },
+
+    testSubscribeToChildrenWithUnnamedParents() {
+        if (!isNodeProccess) {
+            return;
+        }
+        return verifySubscriptionWithParents("@links.Parent.child");
+    },
+
+    testSubscribeToChildrenWithMalformedInclusion1() {
+        if (!isNodeProccess) {
+            return;
+        }
+        return new Promise((resolve, reject) => {
```

Comment:
`return verifySubscriptionWithParents("something.wrong").then(() => throw new Error('subscription should have failed'), (err) => TestCase.assertEqual(err.message, "No property 'something' on object of type 'ObjectA'"));`